pkg/datapath/maps/map.go

```diff
@@ -175,6 +175,10 @@ func (ms *MapSweeper) RemoveDisabledMaps() {
 		}...)
 	}
 
+	if option.Config.KubeProxyReplacement == option.KubeProxyReplacementDisabled {
```

Comment:
This is not enough:
- The SNAT v4 map can be used by `EnableBPFMasquerade && EnableIPv4Masquerade`.
- It is not used when `option.Config.KubeProxyReplacement == option.KubeProxyReplacementPartial && option.Config.EnableNodePort == false`.

Reply:
I've addressed the required changes.